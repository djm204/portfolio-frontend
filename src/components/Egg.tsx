import React, { useState } from 'react'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import cnames from 'classnames'

const Egg = () => {
  const [eggClicks, setEggClicks] = useState(0)

  const eggCounter = () => {
    let currentEggClicks = eggClicks

    if (currentEggClicks < 5) {
      currentEggClicks++
      setEggClicks(currentEggClicks)
    } else {
      const eggClassToSelect = selectEggClassAtRandom(4)
      let eggElement = document.querySelector('.' + eggClassToSelect)

      eggElement?.classList.remove('Egg__Hide')
      eggElement?.classList.add('Egg__Show')

      setTimeout(() => {
        eggElement?.classList.add('Egg__Hide')
        eggElement?.classList.remove('Egg__Show')
      }, 5000)

      setEggClicks(0)
    }
  }

  const selectEggClassAtRandom = (eggCount: number): string => {
    const eggNumber = Math.floor(Math.random() * eggCount) + 1
    return 'Egg--' + eggNumber
  }

  return (
    <React.Fragment>
      <div className="EggBert" onClick={eggCounter}>
        <FontAwesomeIcon icon={'egg'} />
      </div>
      <div className="Counter"></div>
      <div className="Egg--1 Egg__Hide"></div>
      <div className="Egg--2 Egg__Hide"></div>
      <div className="Egg--3 Egg__Hide"></div>
      <div className="Egg--4 Egg__Hide"></div>
    </React.Fragment>
  )
}

export default Egg
