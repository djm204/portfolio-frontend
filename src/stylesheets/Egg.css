.EggBert {
  color: white;
  width: 30px;
  text-align: center;
  font-size: 1em;
  padding: 10px;
  position: absolute;
  right: 20px;
  background-color: transparent;
  border: none;
}

.EggBert:hover {
  color: rgba(255, 211, 0, 0.9);
  cursor: pointer;
}

.Egg img {
  border-radius: 10px;
  width: 70%;
  display: inline-block;
  box-shadow: 2px 3px rgba(0, 0, 0, 0.1);
}

.Egg--1 {
  position: fixed;
  left: -500px;
  top: 10%;
  background-size: cover;
  transition: all 3s ease-in-out;
}

.Egg--1.Egg__Show {
  left: 0px;
}

.Egg--2 {
  width: 500px;
  height: 400px;
  position: fixed;
  left: -500px;
  top: 40%;
  background-size: cover;
  transition: all 3s ease-in-out;
}

.Egg--2.Egg__Show {
  left: 0px;
}

.Egg--3 {
  width: 500px;
  height: 400px;
  position: fixed;
  right: -500px;
  top: 40%;
  background-size: cover;
  transition: all 3s ease-in-out;
}

.Egg--3.Egg__Show {
  right: 0px;
}

.Egg--4 {
  width: 500px;
  height: 400px;
  position: fixed;
  right: -500px;
  top: 10%;
  background-size: cover;
  transition: all 3s ease-in-out;
}

.Egg--4.Egg__Show {
  right: 0px;
}

.Egg__Hide {
  display: inline-block;
}

.Egg__Show {
  display: inline-block;
}
