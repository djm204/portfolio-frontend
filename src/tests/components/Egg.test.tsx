import React from 'react'
import Egg from '../../components/Egg'
import { render, screen, fireEvent } from '@testing-library/react'

it('Egg renders without crashing', async () => {
  const eggRender = render(<Egg />)
  expect(eggRender).toMatchSnapshot()
})

it('Egg shows a picture when clicked', async () => {
  const eggRender = render(<Egg />)
  const egg = eggRender.container

  fireEvent.click(screen.getByRole(/show-eggs/i))
  expect(egg.querySelectorAll('.Egg__Show')).toHaveLength(1)
})

it('Egg shows only 1 picture when clicked repeatedly, quickly.', async () => {
  const eggRender = render(<Egg />)
  const egg = eggRender.container

  //Spam the egg
  for (let i = 0; i < 120; i++) {
    fireEvent.click(screen.getByRole(/show-eggs/i))
  }
  expect(egg.querySelectorAll('.Egg__Show')).toHaveLength(1)
})
