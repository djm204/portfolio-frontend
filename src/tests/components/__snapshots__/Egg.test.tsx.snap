// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Egg renders without crashing 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="EggBert"
        role="show-eggs"
      >
        <i
          class="fa"
        />
      </button>
      <div
        class="Egg Egg--1 Egg__Hide"
        data-testid="egg"
      >
        <div
          class="image-wrapper"
        >
          <img
            alt="I love to festival"
            src="https://davidmendez.dev/i_love_to_festival.jpg"
          />
        </div>
      </div>
      <div
        class="Egg Egg--2 Egg__Hide"
        data-testid="egg"
      >
        <div
          class="image-wrapper"
        >
          <img
            alt="I love to garden"
            src="https://davidmendez.dev/i_love_to_garden.jpg"
          />
        </div>
      </div>
      <div
        class="Egg Egg--3 Egg__Hide"
        data-testid="egg"
      >
        <div
          class="image-wrapper"
        >
          <img
            alt="I love my cats"
            src="https://davidmendez.dev/i_love_my_cats.jpg"
          />
        </div>
      </div>
      <div
        class="Egg Egg--4 Egg__Hide"
        data-testid="egg"
      >
        <div
          class="image-wrapper"
        >
          <img
            alt="My love and I"
            src="https://davidmendez.dev/me_and_my_love.jpg"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <button
      class="EggBert"
      role="show-eggs"
    >
      <i
        class="fa"
      />
    </button>
    <div
      class="Egg Egg--1 Egg__Hide"
      data-testid="egg"
    >
      <div
        class="image-wrapper"
      >
        <img
          alt="I love to festival"
          src="https://davidmendez.dev/i_love_to_festival.jpg"
        />
      </div>
    </div>
    <div
      class="Egg Egg--2 Egg__Hide"
      data-testid="egg"
    >
      <div
        class="image-wrapper"
      >
        <img
          alt="I love to garden"
          src="https://davidmendez.dev/i_love_to_garden.jpg"
        />
      </div>
    </div>
    <div
      class="Egg Egg--3 Egg__Hide"
      data-testid="egg"
    >
      <div
        class="image-wrapper"
      >
        <img
          alt="I love my cats"
          src="https://davidmendez.dev/i_love_my_cats.jpg"
        />
      </div>
    </div>
    <div
      class="Egg Egg--4 Egg__Hide"
      data-testid="egg"
    >
      <div
        class="image-wrapper"
      >
        <img
          alt="My love and I"
          src="https://davidmendez.dev/me_and_my_love.jpg"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
